File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, arr: List[int]) -> int:
        left = 0
        right = len(arr) - 1
        water_cap = 0

        while(left <right ):
            height = min(arr[left], arr[right])
            cap = (right - left) * height
            water_cap = max(water_cap,cap)
            if arr[left] < arr[right]:
                left += 1
            else:
                right -= 1
        return water_cap

        

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.swift
class Solution {
    func search(_ arr: [Int], _ target: Int) -> Int {
        var left = 0
        var right = arr.count - 1
        while(left <= right) {
            let mid = left + (right - left) / 2
            if arr[mid] == target {
                return mid
            } 
            if arr[left] <= arr[mid] {
                if arr[left] <= target && target < arr[mid] {
                    right = mid - 1
                } else {
                    left = mid + 1
                }
            } else {
                if arr[mid] < target && target <= arr[right] {
                    left = mid + 1
                } else {
                    right = mid - 1
                }
            }
        }
        return -1

    }
}
